### PR TITLE
QOL: Allow Modifiers to be shown by prefix or suffix

### PR DESCRIPTION
**Pull request by pHiney**
Partial fix of [Crafting Bench modifiers do not respect any of the crafting rules](https://feathub.com/LocalIdentity/PathOfBuilding/+41) . The rest of the request appears to have been solved by adding the '(Prefix) and '(Suffix)' to the modifier list. This should complete the request.

### Description of the problem being solved:
Crafting Modifiers prefix or suffix are joined together.

### Steps taken to verify a working solution:
- Open a build. 
- Edit an item
- Select "Add Modifier ..."
- Choose a Source and Modifier type
- Choose you Modifiers and select Add

### Before screenshot:
image

### After screenshot:
The default is still to display all Modifiers
image
(edited picture, you can't have both dropdowns open together)

Selecting prefixes shows only prefixes, suffixes does what you think. The '(Prefix) and '(Suffix) is removed as they are redundant.
image
image

The Modifier DropDown disappears when selecting a Source of 'Prefix', 'Suffix', or 'Custom' - and will reappear when other sources are selected.
image
image

### CODE REVIEW

**Comment by pHiney** [comment]
Need to review names of variables

I think the 'local function addModifier()' should be moved up to under buildMods so as to not have functions in the middle of code.

pH

File: src/Classes/ItemsTab.lua [hunk 1]
```diff
@@ -2191,16 +2215,25 @@ function ItemsTabClass:AddCustomModifierToDisplayItem()
 				end
 				return modA.level > modB.level
 			end)
-		elseif sourceId == "VEILED" then
+		elseif currentSourceId == "VEILED" then
 			for i, mod in pairs(self.build.data.veiledMods) do
 				if self.displayItem:GetModSpawnWeight(mod) > 0 then
-					t_insert(modList, {
-						label =  table.concat(mod, "/") .. " (" .. mod.type .. ")",
+					local currMod = {
```
Comment (Wires77) [inline src/Classes/ItemsTab.lua:2221]: You don't need the big if statement and creation of a local table here.  You can just add a ternary to the statement that was previously there, i.e.
```lua
t_insert(modList, {
	label =  table.concat(mod, "/") .. currentModId == "MODIFIERS" and " (" .. mod.type .. ")" or "",
```

File: src/Classes/ItemsTab.lua [hunk 2]
```diff
@@ -2266,14 +2309,28 @@ function ItemsTabClass:AddCustomModifierToDisplayItem()
...
-	controls.modSelectLabel = new("LabelControl", {"TOPRIGHT",nil,"TOPLEFT"}, 95, 45, 0, 16, "^7Modifier:")
-	controls.modSelect = new("DropDownControl", {"TOPLEFT",nil,"TOPLEFT"}, 100, 45, 600, 18, modList)
+	controls.modSelectLabel = new("LabelControl", {"TOPRIGHT",nil,"TOPLEFT"}, 100, 45, 0, 16, "^7Modifier:")
+	controls.modSelectLabel.shown = function()
+		local sourceId = sourceList[controls.source.selIndex].sourceId
+		return sourceId == "CUSTOM" or sourceId == "PREFIX" or sourceId == "SUFFIX"
+	end
+	controls.modSelectDropDown = new("DropDownControl", {"TOPRIGHT",nil,"TOPLEFT"}, 99, 45, 90, 18, modifiersDDList , function(index, value)
+		currentModId = value.modId
+		buildMods()
```
Comment (Wires77) [inline src/Classes/ItemsTab.lua:2326]: You really don't need to change the method signature everywhere just to enable this call.  Just grab the sourceId from the other control and add the modId as a second parameter:

```suggestion
		buildMods(controls.source.list[controls.source.selIndex].sourceId, value.modId)
```

**Review by Wires77 — CHANGES_REQUESTED** [review]
Thanks for the contribution, but I'm not certain this change is necessary, as the feature request you linked to is pretty old at this point and I haven't seen much need for additional filtering outside of what is available.  It seems like a lot extra code complexity for not a whole lot of gain.  I had some code comments before coming to this conclusion, so I'll include them still, but overall I'm inclined to close this.

**Pull request closed without merging**